src/vcUnitConversion.cpp

```diff
@@ -0,0 +1,174 @@
+#include "vcUnitConversion.h"
+
+double vcUnitConversion_ConvertDistance(double sourceValue, vcDistanceUnit sourceUnit, vcDistanceUnit requiredUnit)
+{
+  double scalar = 1.0;
+
+  // Metric
+  if (sourceUnit == vcDistance_Metres)
+    scalar = 1.0;
+  else if (sourceUnit == vcDistance_Kilometres) // If source unit is in Meter than Multiply by 1000 to get kilometer.
+    scalar = 1000;
+  else if (sourceUnit == vcDistance_Centimetres) // If source unit is in Meter than divide by 100 to get Centimeter.
+    scalar = 1 / 100;
+  else if (sourceUnit == vcDistance_Millimetres) // If source unit is in Meter than divide by 1000. to get Millimetres.
+    scalar = 1 / 1000;
+
+  // Imperial
+  else if (sourceUnit == vcDistance_USSurveyFeet) // If source unit is in Meter than divide by 3.28 to get FEET.
+    scalar = 1 / 3.28;
+  else if (sourceUnit == vcDistance_USSurveyMiles)  // If source unit is in miles than divide by 1609.34 to get meter.
+    scalar = 1 / 1609.34;
+  else if (sourceUnit == vcDistance_USSurveyInches)   // If source unit is in Meter than divide by 39.37 to get Inches.
+    scalar = 1 / 39.37;
+  // Nautical
+  else if (sourceUnit == vcDistance_NauticalMiles) // If source unit is in Meter than divide by 1852 to get NAUTICALMILES.
+    scalar = 1 / 1852;
+
+  // Metric
+  if (requiredUnit == vcDistance_Metres)
+    scalar = 1.0;
+  else if (requiredUnit == vcDistance_Kilometres)   //If require unit is in Meter than divide by 1000 to get kilometer.
+    scalar = 1 / 1000;
+  else if (requiredUnit == vcDistance_Centimetres)  //Here one meter is equal to 100 Centimeter.
+    scalar = 100;
+  else if (requiredUnit == vcDistance_Millimetres) // 1 Meter = 10000 millimeter.
+    scalar = 1000;
+
+  // Imperial_US
+  else if (requiredUnit == vcDistance_USSurveyFeet) //  1 meter = 3.28 Feet
+    scalar = 3.28;
+  else if (requiredUnit == vcDistance_USSurveyMiles)   //1 mile = 1609.34 meter
+    scalar = 1609.34;
+  else if (requiredUnit == vcDistance_USSurveyInches)
+    scalar = 39.37;
+
+  // Nautical
+  else if (sourceUnit == vcDistance_NauticalMiles)
```

Comment:
Did you intend `requiredUnit == vcDistance_NauticalMiles` here?

Reply:
Yup, its nautical miles.